### PR TITLE
zh-TW: cr2 Day 3 translations

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all zh-TW content is reviewed.

#1463 #684

### CODE REVIEW

File: po/zh-TW.po [hunk 1]
```diff
@@ -7109,73 +7109,73 @@ msgstr "歡迎參加第 3 天課程"
...
 
 #: src/memory-management.md
 msgid "[Clone](./memory-management/clone.md) (2 minutes)"
-msgstr ""
+msgstr "[複製](./memory-management/clone.md) (2 分鐘)"
 
 #: src/memory-management.md
 msgid "[Copy Types](./memory-management/copy-types.md) (5 minutes)"
-msgstr ""
+msgstr "[複製型別](./memory-management/copy-types.md) (5 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:7166]:
```suggestion
msgstr "[拷貝 (Copy) 型別](./memory-management/copy-types.md) (5 分鐘)"
```

File: po/zh-TW.po [hunk 2]
```diff
@@ -7622,26 +7639,30 @@ msgstr ""
...
 msgid "Clone"
-msgstr ""
+msgstr "複製"
 
 #: src/memory-management/clone.md
 msgid ""
 "Sometimes you _want_ to make a copy of a value. The `Clone` trait "
 "accomplishes this."
-msgstr ""
+msgstr "有時候，您可能會「想要」__複製值。`Clone` 特徵可完成這項作業。"
```
Comment (mingyc) [inline po/zh-TW.po:7648]:
```suggestion
msgstr "有時候，您可能會「想要」複製一個值。`Clone` 特徵可完成這項作業。"
```

File: po/zh-TW.po [hunk 3]
```diff
@@ -8282,6 +8371,8 @@ msgid ""
 "In the example, `pets` is allocated on the stack and the vector data is on "
 "the heap. The two vector elements are _fat pointers_:"
 msgstr ""
+"在本例中，`pets` 和向量資料分別在堆疊和堆積上分配。這兩個向量元素都是「虛指"
+"標」__："
```
Comment (mingyc) [inline po/zh-TW.po:8375]:
```suggestion
"標」："
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -8453,6 +8552,8 @@ msgid ""
 "Note that the requirement is that conflicting references not _exist_ at the "
 "same point. It does not matter where the reference is dereferenced."
 msgstr ""
+"請注意，這裡的規定是同一點上不得「存在」__衝突的參照，在何處解除參照並不重"
```
Comment (mingyc) [inline po/zh-TW.po:8555]:
```suggestion
"請注意，這裡的規定是同一點上不得「存在」衝突的參照，在何處解除參照並不重"
```

File: po/zh-TW.po [hunk 5]
```diff
@@ -8485,6 +8586,9 @@ msgid ""
 "optimize code. For example, a value behind a shared reference can be safely "
 "cached in a register for the lifetime of that reference."
 msgstr ""
+"專屬參照的約束力很強。Rust 會利用這類參照，確保資料競爭的情形不會發生；此外，"
+"也會「透過」__這項約束，將程式碼最佳化。舉例來說，共用參照背後的值可以在該參"
```
Comment (mingyc) [inline po/zh-TW.po:8590]:
```suggestion
"也會「透過」這項約束，將程式碼最佳化。舉例來說，共用參照背後的值可以在該參"
```

File: po/zh-TW.po [hunk 6]
```diff
@@ -8515,30 +8626,33 @@ msgstr "`RefCell<T>`"
...
+"`Cell` 會納入值，並允許取得或設定該值，即使具有對 `Cell` 的共用參照也一樣。但"
+"是，它不允許對該值進行任何參照。由於沒有參照，因此借用規則不得違反。"
 
 #: src/borrowing/interior-mutability.md
 msgid ""
 "The main thing to take away from this slide is that Rust provides _safe_ "
 "ways to modify data behind a shared reference. There are a variety of ways "
 "to ensure that safety, and `RefCell` and `Cell` are two of them."
 msgstr ""
+"這張投影片的重點是 Rust 提供「安全的」__方法，可讓您修改共用參照背後的資料。"
```
Comment (mingyc) [inline po/zh-TW.po:8654]:
```suggestion
"這張投影片的重點是 Rust 提供「安全的」方法，可讓您修改共用參照背後的資料。"
```

File: po/zh-TW.po [hunk 7]
```diff
@@ -8827,7 +8954,7 @@ msgstr "位元組字串可讓您直接建立 `&[u8]` 值："
 msgid ""
 "A reference has a _lifetime_, which must not \"outlive\" the value it refers "
 "to. This is verified by the borrow checker."
-msgstr ""
+msgstr "參照的「生命週期」__不得「超過」其所參照的值。此由借用檢查器負責驗證。"
```
Comment (mingyc) [inline po/zh-TW.po:8957]:
```suggestion
msgstr "參照的「生命週期」不得「超過」其所參照的值。此由借用檢查器負責驗證。"
```

File: po/zh-TW.po [hunk 8]
```diff
@@ -8944,6 +9083,10 @@ msgid ""
 "help resolve ambiguity. Often, especially when prototyping, it's easier to "
 "just work with owned data by cloning values where necessary."
 msgstr ""
+"學生可能會詢問何時該使用生命週期。Rust 的借用「一律」__具有生命週期。在大多數"
```
Comment (mingyc) [inline po/zh-TW.po:9086]:
```suggestion
"學生可能會詢問何時該使用生命週期。Rust 的借用「一律」具有生命週期。在大多數"
```

File: po/zh-TW.po [hunk 9]
```diff
@@ -8312,68 +8404,74 @@ msgid ""
 "value. For a given node N, all nodes in a N's left subtree contain smaller "
 "values, and all nodes in N's right subtree will contain larger values."
 msgstr ""
+"二元樹是一種樹狀資料結構，其中每個節點都有左右兩個子項。我們會建立每個節點都"
```
Comment (mingyc) [inline po/zh-TW.po:8407]:
```suggestion
"二元樹是一種樹狀資料結構，其中每個節點都有左右兩個子節點。我們會建立每個節點都"
```

File: po/zh-TW.po [hunk 10]
```diff
@@ -8493,20 +8597,27 @@ msgid ""
 "time. But, there are some situations where it doesn't quite \"get it\" and "
 "this often results in \"fighting with the borrow checker.\""
 msgstr ""
+"借用檢查器在設計上涵蓋了許多常見模式，例如同時對結構體中的不同欄位進行專屬參"
+"照。但可能也是有某些無法完全「理解」的情況，這通常會導致「與借用檢查器衝"
```
Comment (mingyc) [inline po/zh-TW.po:8601]:
```suggestion
"照。但檢查器也可能無法完全「理解」某些的情況，這通常會導致「與借用檢查器衝"
```

File: po/zh-TW.po [hunk 11]
```diff
@@ -8493,20 +8597,27 @@ msgid ""
 "time. But, there are some situations where it doesn't quite \"get it\" and "
 "this often results in \"fighting with the borrow checker.\""
 msgstr ""
+"借用檢查器在設計上涵蓋了許多常見模式，例如同時對結構體中的不同欄位進行專屬參"
```
Comment (mingyc) [inline po/zh-TW.po:8600]:
```suggestion
"借用檢查器在設計上考量了許多常見模式，例如同時對結構體中的不同欄位進行專屬參"
```

File: po/zh-TW.po [hunk 12]
```diff
@@ -8493,20 +8597,27 @@ msgid ""
...
+"照。但可能也是有某些無法完全「理解」的情況，這通常會導致「與借用檢查器衝"
+"突」。"
 
 #: src/borrowing/interior-mutability.md
 msgid ""
 "In some situations, it's necessary to modify data behind a shared (read-"
 "only) reference. For example, a shared data structure might have an internal "
 "cache, and wish to update that cache from read-only methods."
 msgstr ""
+"在某些情況下，您必須修改共用 (唯讀) 參照背後的資料。比方說，共用的資料結構可"
```
Comment (mingyc) [inline po/zh-TW.po:8610]:
```suggestion
"在某些情況下，您必須修改共用 (唯讀) 參照背後的資料：比方說，共用的資料結構可"
```

File: po/zh-TW.po [hunk 13]
```diff
@@ -8493,20 +8597,27 @@ msgid ""
...
+"能含有內部快取，並想透過唯讀方法更新該快取，就屬於這種情況。"
```
Comment (mingyc) [inline po/zh-TW.po:8611]:
```suggestion
"能含有內部快取，並想透過唯讀方法更新該快取。"
```

**Review by mingyc — CHANGES_REQUESTED** [review]

File: po/zh-TW.po [hunk 14]
```diff
@@ -7387,24 +7404,25 @@ msgid ""
 "garbage collector starts with a set of \"roots\" to find all reachable "
 "memory. Rust's \"single owner\" principle is a similar idea."
 msgstr ""
+"熟悉垃圾收集實作的學員會知道，垃圾收集器是從一組「根」開始尋找所有可存取的記"
```
Comment (mingyc) [inline po/zh-TW.po:7407]:
```suggestion
"熟悉垃圾回收實作的學員會知道，垃圾回收器是從一組「根」開始尋找所有可存取的記"
```

File: po/zh-TW.po [hunk 15]
```diff
@@ -7622,26 +7639,30 @@ msgstr ""
 
 #: src/memory-management/clone.md
 msgid "Clone"
-msgstr ""
+msgstr "複製"
```
Comment (mingyc) [inline po/zh-TW.po:7642]:
```suggestion
msgstr "`Clone` 複製"
```

**Review by mingyc — CHANGES_REQUESTED** [review]

File: po/zh-TW.po [hunk 16]
```diff
@@ -8849,39 +8978,44 @@ msgstr "生命週期註解會建立限制；編譯器會驗證是否有有效的
...
-msgstr ""
+msgstr "// What is the lifetime of p3?\n"
 
 #: src/slices-and-lifetimes/lifetime-annotations.md
 msgid "\"p3: {p3:?}\""
-msgstr ""
+msgstr "\"p3: {p3:?}\""
 
 #: src/slices-and-lifetimes/lifetime-annotations.md
+#, fuzzy
```
Comment (henrif75) [inline po/zh-TW.po:8992]: Remove

**Comment by mingyc** [comment]
There are some formatting issues.

**Review by mingyc — APPROVED** [review]

**Comment by henrif75** [comment]
@mingyc Thank you!

**Pull request merged**